src/producer/messageProducer.js

```diff
@@ -56,13 +56,25 @@ module.exports = ({ logger, cluster, partitioner, eosManager, idempotent, retrie
       }
     }
 
+    const mergedTopicMessages = topicMessages.reduce((merged, { topic, messages }) => {
+      const topicBatch = merged.find(({ topic: mergedTopic }) => topic === mergedTopic)
+
+      if (!topicBatch) {
+        merged.push({ topic, messages })
+      } else {
+        topicBatch.messages.concat(messages)
```

Comment:
surprisingly enough, the spread operation is faster than concat.

`topicBatch.messages = [...topicBatch.messages, ...messages]`

Reply:
Totally bike shedding here, but I wonder whether that's because concatenation to any empty array is faster 🧐. Doesn't matter, move on, nothing to see here 😇.